src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlCompositeMemberTranslator.cs

```diff
@@ -50,9 +50,18 @@ public class NpgsqlCompositeMemberTranslator : RelationalCompositeMemberTranslat
             [NotNull] INpgsqlOptions npgsqlOptions)
             : base(dependencies)
         {
-            // ReSharper disable once VirtualMemberCallInConstructor
+            var instanceTranslators =
+                new IMemberTranslator[]
+                {
+                    new NpgsqlArrayMemberTranslator(npgsqlOptions.PostgresVersion)
+                };
+
+            // ReSharper disable once DoNotCallOverridableMethodsInConstructor
             AddTranslators(MemberTranslators);
 
+            // ReSharper disable once DoNotCallOverridableMethodsInConstructor
+            AddTranslators(instanceTranslators);
```

Comment:
nit: naming isn't ideal: `MemberTranslators` vs `instanceTranslators`. Maybe rename to `versionDependentTranslators`?